src/server/utils/parseDocument.ts

```diff
@@ -39,13 +39,21 @@ export function parseDocument($: cheerio.Root): ParsedDocument {
 
       let $sectionElements;
       if ($h.is($pageTitle)) {
-        const $firstElement = $("article")
-          .children() // div.markdown, header
-          .not("header") // div.markdown
-          .children() // h1, p, p, h2, ...
-          .first(); // h1 || p
-        if ($firstElement.filter(HEADINGS).length > 0) {
-          return;
+        let $firstElement = $("article").children().first();
+
+        if ($firstElement.filter("header").length > 0) {
+          $firstElement = $("article")
+            .children() // div.markdown, header
+            .not("header") // div.markdown
+            .children() // h1, p, p, h2, ...
+            .first(); // h1 || p
+          if ($firstElement.filter(HEADINGS).length > 0) {
+            return;
+          }
+        } else {
+          $("article > .markdown")
+            .children() // h1, p, p, h2, ...
+            .first(); // h1 || p
```

Comment:
I think these lines are doing nothing?
maybe this should be assigned to `$firstElement`?

```suggestion
          $firstElement = $("article > .markdown")
            .children() // h1, p, p, h2, ...
            .first(); // h1 || p
```

Reply:
You're right! I added this branch to handle the tests for the older layout but I thought I added a test for this case too. I will investigate why the tests passes with this.

Reply:
I misunderstood this code a bit, I think this is not needed now when everything is the same for the main heading. Also added a proper test for it this time :)